src/library.cpp

```diff
@@ -277,11 +278,13 @@ std::vector<std::string> Library::listBooksIds(
       std::set<std::string> sBookTags(vBookTags.begin(), vBookTags.end());
       bool ok = true;
       for (auto& t: tags) {
-        if (sBookTags.find(t) == sBookTags.end()) {
+        if ((!other && sBookTags.find(t) == sBookTags.end()) || (other && sBookTags.find(t) != sBookTags.end())) {
           // A "filter" tag is not in the book tag.
           // No need to loop for all "filter" tags.
           ok = false;
-          break;
+          if (!other) {
+            break;
+          }
```

Comment:
We don't need to check `other` here.
If `ok` is set to false, it can never be made true again. So we don't need to continue the loop. Whatever the value of `other` is.